### ISSUE TITLE
Subscribing to topic keep looping through CONNECT

### ISSUE THREAD

**Issue by dhabib29** [eclipse/mosquitto#894]
Hi,
I am pretty new to MQTT, however after completing the installation on Raspberrypi3 running on Jessie when I am trying to test mqtt if keep trying to connect what it means is after i run the below command:
mosquitto_sub -d -u username -P password -t "dev/test"
the response I am getting is Client mosqsub/1017-pi sending CONNECT and this keep repeating like it is in some kind of loop.
I also checked the systemctl status mosquitto.service and it is showing as active.
I also tried uninstalling and reinstalling but issue is always the same and log file has nothing in it even I have log_type all in the config file.
your help is appreciated.

**Comment by toast-uz** [comment]
Your log was entirely empty? Didn't you have any startup message?

**Comment by dhabib29** [comment]
yes just the startup message and when I restart pi closing message

**Comment by dhabib29** [comment]
1532374056: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting
1532374056: Config loaded from /etc/mosquitto/mosquitto.conf.
1532374056: Opening ipv4 listen socket on port 1889.
1532374056: Opening ipv6 listen socket on port 1889.
1532374193: Error in poll: Interrupted system call.
1532374193: mosquitto version 1.4.10 terminating
1532374200: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting
1532374200: Config loaded from /etc/mosquitto/mosquitto.conf.
1532374201: Opening ipv4 listen socket on port 1889.
1532374201: Opening ipv6 listen socket on port 1889.
1532376148: Saving in-memory database to /var/lib/mosquitto/mosquitto.db.
1532377155: Error in poll: Interrupted system call.
1532377155: mosquitto version 1.4.10 terminating
1532377162: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting
1532377162: Config loaded from /etc/mosquitto/mosquitto.conf.
1532377162: Opening ipv4 listen socket on port 1889.
1532377162: Opening ipv6 listen socket on port 1889.
1532385617: Error in poll: Interrupted system call.
1532385617: mosquitto version 1.4.10 terminating
1532385626: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting


this is whats in the log

**Comment by toast-uz** [comment]
A lot of information !!
- Your mosquitto's version 1.4.10 is too old. Could you upgrade?
- Your mosquitto binds on port 1889. You had some customization on the mosquitto,conf . Could you show it?
- Your mosquitto stopped abnormally.

**Comment by toast-uz** [comment]
p.s.

- Your mosquitto_sub's target port is default (= 1883), while your mosquitto broker binds on port 1889. Why do you think they can connect with each other?

**Comment by dhabib29** [comment]
I tried the latest installation using sudo apt-get install mosquitto. yes I only changed the port in the config file to check if that is the port issue however with changing the port or keeping it to default doesn't make any difference.
How to upgrade to latest version?
I dont know why it is stopping abnormally I am restarting the pi using sudo reboot.

**Comment by toast-uz** [comment]
On current version, clear current logs, clear mosquitto.conf without log_type all, reproduce your issue, and show the new logs.
I'm afraid I have no expertise on building specific environment. You can ask such questions at mosquitto-dev ML.

**Comment by dhabib29** [comment]
1532443852: Error in poll: Interrupted system call.
1532443852: mosquitto version 1.4.10 terminating
1532443853: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting
1532443853: Config loaded from /etc/mosquitto/mosquitto.conf.
1532443853: Opening ipv4 listen socket on port 1883.
1532443853: Error: Address already in use

I did all of that, i also cahnge the port to default 1883 above is the log

**Comment by toast-uz** [comment]
Is there any log prior of these logs? Mosquitto cannot terminate before starting. Please don't trim the information.

**Comment by dhabib29** [comment]
no there is nothing before that.
after cleaning the log and changing the config file I restarted mosquitto. and this what get looged.

**Comment by toast-uz** [comment]
Could you restart pi itself and show your logs? You had such a "ghost" mosquitto, so mosquitto failed restarting.

**Comment by dhabib29** [comment]
1532443852: Error in poll: Interrupted system call.
1532443852: mosquitto version 1.4.10 terminating
1532443853: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting
1532443853: Config loaded from /etc/mosquitto/mosquitto.conf.
1532443853: Opening ipv4 listen socket on port 1883.
1532443853: Error: Address already in use
1532444784: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting
1532444784: Config loaded from /etc/mosquitto/mosquitto.conf.
1532444784: Opening ipv4 listen socket on port 1883.
1532444784: Error: Address already in use
1532444846: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting


done the restart.
above is the log file.
I really appreciate your help.

**Comment by toast-uz** [comment]
This issue definitely comes from your broker starting abnormally.
Could you change the subject of this issue to such a "Error: Address already in use on pi" in order to describe the issue clearly?

This error "Address already in use" seems similar to #825 .
Try the workaround deleting the line "listener 1883" in /etc/mosquitto/conf.d/pi.conf .

**Comment by dhabib29** [comment]
it seems like I don't have the pi.conf in conf.d folder
however i commented on "listener 1883" by doing that already in use error goes away however error in poll: interrupted system call remains 

1532443934: Error in poll: Interrupted system call.
1532443934: mosquitto version 1.4.10 terminating
1532443934: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting

**Comment by toast-uz** [comment]
Did you restart pi itself? By the last line of your logs, mosquitto seemed starting normally.

**Comment by toast-uz** [comment]
Sorry if your mosquitto starts normally, there should be the logs as followings:

``
Config loaded from /etc/mosquitto/mosquitto.conf.
Opening ipv4 listen socket on port 1883.
```

**Comment by dhabib29** [comment]
Yes it did restarted the pi and mosquitto.
but the original issue remains. which is CONNECT keep going through the loop.

**Comment by toast-uz** [comment]
Sorry I cannot support the issue depends on pi. Wait someone can support.

**Comment by dhabib29** [comment]
No problem. I will wait.
Thanks toast for all your help.

**Comment by ralight** [comment]
Could you please try installing from the mosquitto repository? There are details on how to do that here: https://www.digitalocean.com/community/tutorials/how-to-install-and-secure-the-mosquitto-mqtt-messaging-broker-on-debian-8

**Comment by alwashe** [comment]
i think i have the same Problem

installation on a Raspberry Pi 3b, where i have Raspbian 4.14, mosquito and mosquito-clienets. I upgraded after I got the error from 1.4.10 to 1.5.3. But this did not solve my problem.

i can connect

My Error:
`   
pi@mqtt:~ $ mosquitto -v
1539077460: mosquitto version 1.5.3 starting
1539077460: Using default config.
1539077460: Opening ipv4 listen socket on port 1883.
1539077460: Error: Address already in use
`

Mosquitto Config:
`
pi@mqtt:~ $ cat /etc/mosquitto/mosquitto.conf
# Place your local configuration in /etc/mosquitto/conf.d/
#
# A full description of the configuration file is at
# /usr/share/doc/mosquitto/examples/mosquitto.conf.example

pid_file /var/run/mosquitto.pid

persistence true
persistence_location /var/lib/mosquitto/

log_dest file /var/log/mosquitto/mosquitto.log
`

log:
`pi@raspberrypi:~ $ cat /var/log/mosquitto/mosquitto.log
1539079636: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting
1539079636: Config loaded from /etc/mosquitto/mosquitto.conf.
1539079636: Opening ipv4 listen socket on port 1883.
1539079636: Opening ipv6 listen socket on port 1883.
1539079649: Error in poll: Interrupted system call.
1539079649: mosquitto version 1.4.10 terminating
1539079654: mosquitto version 1.4.10 (build date Fri, 22 Dec 2017 08:19:25 +0000) starting
1539079654: Config loaded from /etc/mosquitto/mosquitto.conf.
1539079654: Opening ipv4 listen socket on port 1883.
1539079654: Opening ipv6 listen socket on port 1883.
1539081462: Saving in-memory database to /var/lib/mosquitto/mosquitto.db.
1539081987: New connection from 192.168.178.70 on port 1883.
`
